### PR TITLE
bowang-updated with 4 new components: alert, menu, pagination, switch

### PR DESCRIPTION
**Pull request by bowang-ynd**
I have updated with 4 new components: alert, menu, pagination, switch; and tested the dash-mui locally and worked well.

I plan to add a few more in this week.

### CODE REVIEW

File: demo/pages/menu.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        },
+    ),
+]
+
+# Create the layout using the menus defined above
+layout = html.Div(
+    mui.Grid(
+        spacing=2,  # Controls the spacing between the menus
+        children=[
+            create_component_item(title, mui.Menu, props) for title, props in menus
```
Comment (LatentDream) [inline demo/pages/menu.py:41]: Am I missing something ? Running the demo, this line return an Error:
```
(review-py3.12) [work] dream@Latent:~/repo/dash-mui.git/review/demo$ python app.py

Traceback (most recent call last):
  File "/home/dream/repo/dash-mui.git/review/demo/app.py", line 10, in <module>
    app = Dash(__name__, use_pages=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dream/.cache/pypoetry/virtualenvs/review-aK06w5nS-py3.12/lib/python3.12/site-packages/dash/dash.py", line 580, in __init__
    self.init_app()
  File "/home/dream/.cache/pypoetry/virtualenvs/review-aK06w5nS-py3.12/lib/python3.12/site-packages/dash/dash.py", line 659, in init_app
    self.enable_pages()
  File "/home/dream/.cache/pypoetry/virtualenvs/review-aK06w5nS-py3.12/lib/python3.12/site-packages/dash/dash.py", line 2189, in enable_pages
    _import_layouts_from_pages(self.config.pages_folder)
  File "/home/dream/.cache/pypoetry/virtualenvs/review-aK06w5nS-py3.12/lib/python3.12/site-packages/dash/_pages.py", line 444, in _import_layouts_from_pages
    spec.loader.exec_module(page_module)
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/home/dream/repo/dash-mui.git/review/demo/pages/menu.py", line 41, in <module>
    create_component_item(title, mui.Menu, props) for title, props in menus
                                 ^^^^^^^^
AttributeError: module 'dash_material_ui' has no attribute 'Menu'


```

Comment (edublancas) [reply demo/pages/menu.py:41]: you're missing some steps, see https://github.com/ploomber/dash-mui/issues/3

I was able to run the demo successfully

Comment (LatentDream) [reply demo/pages/menu.py:41]: Oh, thank you. Reviewing the PR right now

**Review by LatentDream — COMMENTED** [review]
Look Good, beside a little bug with the Menu: 
> When you click on a Menu item that wasn't previously clicked, there is a second Menu being rendered. See video:

https://github.com/user-attachments/assets/4f9f48ba-d55e-4522-b7c9-79bfd3037ed8

**Comment by bowang-ynd** [comment]
looks funny for the second menu to be rendered and pop up. I will fix it shortly!

**Comment by bowang-ynd** [comment]
@LatentDream My latest commit should resolve the second menu being rendered issue.

File: src/lib/components/Menu.react.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                disablePortal  // This ensures the menu is rendered within the same DOM tree
+                {...other}
+            >
+                {menuItems.map((item, index) => (
+                    <MenuItem
+                        key={index}
+                        onClick={(event) => handleMenuItemClick(event, index)}
+                        disabled={item.disabled || false}  // Handle disabled state if the item is an object
+                    >
+                        {typeof item === 'object' ? item.label : item}  {/* Handle either string or object */}
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:59]:
```suggestion
                        {typeof item === 'object' ? item.label : item}
```

File: src/lib/components/Menu.react.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                open={Boolean(anchorEl)}  // Menu is open when anchorEl is not null
+                onClose={handleClose}
+                disablePortal  // This ensures the menu is rendered within the same DOM tree
+                {...other}
+            >
+                {menuItems.map((item, index) => (
+                    <MenuItem
+                        key={index}
+                        onClick={(event) => handleMenuItemClick(event, index)}
+                        disabled={item.disabled || false}  // Handle disabled state if the item is an object
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:57]:
```suggestion
                        disabled={item.disabled || false}
```

File: src/lib/components/Menu.react.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);  // Close the menu
+    };
+
+    const handleMenuItemClick = (event, index) => {
+        if (onItemClick) onItemClick(event, index);  // Trigger callback on item click
+        if (setProps) setProps({ selectedIndex: index });  // Update Dash's selected index state
+        handleClose();  // Close the menu after an item is selected
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:32]:
```suggestion
        handleClose();
```

File: src/lib/components/Menu.react.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        setAnchorEl(event.currentTarget);  // Set the anchor element to open the menu
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);  // Close the menu
+    };
+
+    const handleMenuItemClick = (event, index) => {
+        if (onItemClick) onItemClick(event, index);  // Trigger callback on item click
+        if (setProps) setProps({ selectedIndex: index });  // Update Dash's selected index state
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:31]:
```suggestion
        if (setProps) setProps({ selectedIndex: index });
```

File: src/lib/components/Menu.react.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    const handleButtonClick = (event) => {
+        setAnchorEl(event.currentTarget);  // Set the anchor element to open the menu
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);  // Close the menu
+    };
+
+    const handleMenuItemClick = (event, index) => {
+        if (onItemClick) onItemClick(event, index);  // Trigger callback on item click
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:30]:
```suggestion
        if (onItemClick) onItemClick(event, index);
```

File: src/lib/components/Menu.react.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    ...other
+}) => {
+    const [anchorEl, setAnchorEl] = useState(null);
+
+    const handleButtonClick = (event) => {
+        setAnchorEl(event.currentTarget);  // Set the anchor element to open the menu
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);  // Close the menu
```
Comment (LatentDream) [inline src/lib/components/Menu.react.js:26]:
```suggestion
        setAnchorEl(null);
```

**Review by LatentDream — COMMENTED** [review]
Two minor point:

1. When building the project, a julia file `src/DashMaterialUi.jl` is generetated. 
    - Should this be include in the source ? 
    - Otherwise please add this to the `.gitignore`

2. There is a lot of inline comment in the Menu.react.js, which could be remove. If needed, you could have a single comment on top (line 19) with something like. That way the "doc" is at one place, and not all over:
```
  // `anchorEl` is used as the target where the dropdown will be 
  // displayed. Once the button is clicked, we attach the MenuList 
  // to this  menu, thus triggering the MenuList to appear since it
  // now has a parent.

```

**Comment by edublancas** [comment]
@LatentDream we're not including any aut-generated files in the repo

**Comment by LatentDream** [comment]
> @LatentDream we're not including any aut-generated files in the repo

Okok, I'm not sure about the build system for building process of dash component, thus why I'm asking. Good to know :)

**Comment by bowang-ynd** [comment]
Thanks for spotting it! I will clear the comments and add any update the .gitignore file

**Review by LatentDream — APPROVED** [review]

**Comment by bowang-ynd** [comment]
@LatentDream I have made a new commit and cleared up all in-line comments! I also checked my .gitignore file and it should not include the auto-generated jl file. Thx!

**Comment by edublancas** [comment]
nice work @bowang-ynd!

**Pull request merged**